jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/GitEnricherTest.java

```diff
@@ -191,24 +194,29 @@ void setUp() throws Exception {
         }
 
 
-        Stream<Arguments> controllerResources() {
+        Stream<Arguments> kubernetesResources() {
             return Stream.of(
-              arguments(new ServiceBuilder().build()),
-              arguments(new DeploymentBuilder().build()),
-              arguments(new DeploymentConfigBuilder().build()),
-              arguments(new ReplicaSetBuilder().build()),
-              arguments(new ReplicationControllerBuilder().build()),
-              arguments(new DaemonSetBuilder().build()),
-              arguments(new StatefulSetBuilder().build()),
-              arguments(new JobBuilder().build())
+              arguments(new ServiceBuilder().withNewMetadata().endMetadata().build()),
+              arguments(new DeploymentBuilder().withNewMetadata().endMetadata().build()),
+              arguments(new DeploymentConfigBuilder().withNewMetadata().endMetadata().build()),
+              arguments(new ReplicaSetBuilder().withNewMetadata().endMetadata().build()),
+              arguments(new ReplicationControllerBuilder().withNewMetadata().endMetadata().build()),
+              arguments(new DaemonSetBuilder().withNewMetadata().endMetadata().build()),
+              arguments(new StatefulSetBuilder().withNewMetadata().endMetadata().build()),
+              arguments(new JobBuilder().withNewMetadata().endMetadata().build()),
+              arguments(new SecretBuilder().withNewMetadata().endMetadata().build()),
+              arguments(new ImageStreamBuilder().withNewMetadata().endMetadata().build()),
+              arguments(new BuildConfigBuilder().withNewMetadata().endMetadata().build())
```

Comment:
Note that due to the changes in GitEnricher, now you need to create the ObjectMetaBuilder entry.

I'm not sure that everytime the resources reach the enricher they will contain an ObjectMetaBuilder instance.

Are you sure about this? (consider also fragments and other corner cases)

Reply:
I'm not sure. I checked with a fragment without metadata but I noticed we always seem to initialize metadata while reading fragment into `HasMetadata`:
https://github.com/eclipse/jkube/blob/e101e93ed4bfad99ce9549d85aa968225326839e/jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceFragments.java#L138-L139

Reply:
Using `ObjectMetadataBuilder` also adds git labels in `.metadata` of Pod Template, not sure whether we want to have git annotations there.

Reply:
In the case of resources generated via opinionated defaults, I checked various handlers in `org.eclipse.jkube.kit.enricher.handler` package and they all seem to be generating a non null ObjectMeta always:
https://github.com/eclipse/jkube/blob/e101e93ed4bfad99ce9549d85aa968225326839e/jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/handler/DeploymentHandler.java#L42

Once resources go through enricher's `create` phase, `jkube-metadata` is the first enricher in `enrich` phase where various visitors try to create metadata if it doesn't exist:
https://github.com/eclipse/jkube/blob/e101e93ed4bfad99ce9549d85aa968225326839e/jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/visitor/MetadataVisitor.java#L134